core/src/main/scala/org/apache/spark/FutureAction.scala

```diff
@@ -105,6 +108,7 @@ trait FutureAction[T] extends Future[T] {
  * A [[FutureAction]] holding the result of an action that triggers a single job. Examples include
  * count, collect, reduce.
  */
+@DeveloperApi
```

Comment:
Although we should also consider making them `private[spark]` and adding MiMa excludes, since that's more restrictive than `DeveloperAPI`.